feat: cherry pick (#2)
roel4ez/blog: docs/articles/git-cherry-pick.md

gitGraph
    commit
    branch feature/my-bugfix
    checkout feature/my-bugfix
    commit id: "C1"
    commit id: "C2"
    commit id: "C3"
    commit id: "C4"
    commit id: "C5"
    commit id: "C6"
    checkout main
    merge feature/my-bugfix
```

to this?
    
```mermaid
gitGraph
    commit
    branch feature/my-bugfix-1
    checkout feature/my-bugfix-1
    commit id: "C1"
    checkout main
    branch feature/my-bugfix-2
    checkout feature/my-bugfix-2
    commit id: "C2"
    commit id: "C3"
    commit id: "C4"
    commit id: "C5"
    commit id: "C6"
    checkout main
    merge feature/my-bugfix-1
    merge feature/my-bugfix-2
```

## Show me the code 💻

So do you have to do everything again? **No**, you can use `git cherry-pick` to
setup your branches and create two new PRs.

First, identify the hashes of the commits that are part of the first PR:

```bash title="Identify the commits that are part of the first PR"
$ [feature/my-bugfix] 
    git log --all --decorate --oneline --graph      # Show the commits in this feature branch, in a nicely formatted graph
```

!!! note "Git Commit Hashes"  
    `git commit hashes` are `SHA-1` hashes that contain the commit message, the

>     author, the date, the files that were changed, as well as the parent commit

Reviewer: Actually, it contains _a complete snapshot of all tracked files_:

```suggestion
    author, the date, a complete snapshot of files, as well as the parent commit
```
